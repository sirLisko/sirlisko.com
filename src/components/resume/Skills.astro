---
import type { Resume } from "../../types";

interface Props {
  skills: Resume["skills"];
}

const { skills } = Astro.props;
---

<section class="skills">
  <h2>Skills</h2>
  {
    Object.keys(skills).map((type) => (
      <div>
        <h3>{type}</h3>
        <ul>
          {skills[type as keyof Resume["skills"]].map((skill) => (
            <li>{skill}</li>
          ))}
        </ul>
      </div>
    ))
  }
</section>
