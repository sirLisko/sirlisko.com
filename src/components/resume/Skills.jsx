import React from "react";
import PropTypes from "prop-types";

const Skills = ({ skills }) => (
  <article className="skills">
    <h1>Skills</h1>
    {Object.keys(skills).map(type => (
      <div key={type}>
        <h2>{type}</h2>
        <ul>
          {skills[type].map(skill => (
            <li key={skill}>{skill}</li>
          ))}
        </ul>
      </div>
    ))}
  </article>
);

Skills.propTypes = {
  skills: PropTypes.shape(PropTypes.array.isRequired).isRequired
};

export default Skills;
