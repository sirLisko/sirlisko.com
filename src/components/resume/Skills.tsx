import React from "react";

interface SkillsProps {
  skills: {
    [type: string]: string[];
  };
}

const Skills = ({ skills }: SkillsProps) => (
  <article className="skills">
    <h1>Skills</h1>
    {Object.keys(skills).map((type) => (
      <div key={type}>
        <h2>{type}</h2>
        <ul>
          {skills[type].map((skill) => (
            <li key={skill}>{skill}</li>
          ))}
        </ul>
      </div>
    ))}
  </article>
);

export default Skills;
