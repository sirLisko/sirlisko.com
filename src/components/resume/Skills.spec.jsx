import React from "react";
import renderer from "react-test-renderer";

import Skills from "./Skills.jsx";

describe("Skills Component", () => {
  let wrapper;
  const props = {
    skills: {
      main: ["foo", "bar"],
      "misc.": ["foobar"],
    },
  };

  beforeEach(() => {
    wrapper = renderer.create(<Skills {...props} />).toJSON();
  });

  it("should render properly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
