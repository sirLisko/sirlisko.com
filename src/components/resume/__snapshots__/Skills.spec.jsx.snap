// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skills Component should render properly 1`] = `
<article
  className="skills"
>
  <h1>
    Skills
  </h1>
  <div>
    <h2>
      main
    </h2>
    <ul>
      <li>
        foo
      </li>
      <li>
        bar
      </li>
    </ul>
  </div>
  <div>
    <h2>
      misc.
    </h2>
    <ul>
      <li>
        foobar
      </li>
    </ul>
  </div>
</article>
`;
